toolkit/tools/pkg/imagemodifierlib/modifierutils.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagemodifierlib
+
+import (
+	"strconv"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/imagegen/installutils"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/userutils"
+)
+
+func doModifications(baseConfigPath string, config *imagecustomizerapi.Config) error {
+	err := addOrUpdateUsers(config.SystemConfig.Users, baseConfigPath)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addOrUpdateUsers(users []imagecustomizerapi.User, baseConfigPath string) error {
+	for _, user := range users {
+		err := addOrUpdateUser(user, baseConfigPath)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func addOrUpdateUser(user imagecustomizerapi.User, baseConfigPath string) error {
```

Comment:
It would probably be goodness to figure out a way to avoid needing to duplicate this code.